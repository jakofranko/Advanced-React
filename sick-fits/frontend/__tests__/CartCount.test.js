import { shallow, mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import CartCount from '../components/CartCount';


describe('<CartCount />', () => {
    it('renders', () =>{
        const comp = shallow(<CartCount count={10} />);
        expect(toJSON(comp)).toMatchSnapshot();
    });

    it('updates via props', () => {
        const comp = mount(<CartCount count={10} />);
        expect(toJSON(comp)).toMatchSnapshot();

        comp.setProps({ count: 50 });
        expect(toJSON(comp)).toMatchSnapshot();
    });
});
