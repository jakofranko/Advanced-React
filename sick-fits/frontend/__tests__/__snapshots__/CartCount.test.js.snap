// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartCount /> renders 1`] = `
<CartCount__AnimationStyles>
  <TransitionGroup
    childFactory={[Function]}
    component="div"
  >
    <CSSTransition
      className="count"
      classNames="count"
      key="10"
      timeout={
        Object {
          "enter": 400,
          "exit": 400,
        }
      }
      unmountOnExit={true}
    >
      <CartCount__Dot>
        10
      </CartCount__Dot>
    </CSSTransition>
  </TransitionGroup>
</CartCount__AnimationStyles>
`;

exports[`<CartCount /> updates via props 1`] = `
<CartCounter
  count={10}
>
  <CartCount__AnimationStyles>
    <span
      className="CartCount__AnimationStyles-xxvp4g-0 gDkasO"
    >
      <TransitionGroup
        childFactory={[Function]}
        component="div"
      >
        <div>
          <CSSTransition
            className="count"
            classNames="count"
            in={true}
            key=".$10"
            onExited={[Function]}
            timeout={
              Object {
                "enter": 400,
                "exit": 400,
              }
            }
            unmountOnExit={true}
          >
            <Transition
              appear={false}
              className="count"
              enter={true}
              exit={true}
              in={true}
              mountOnEnter={false}
              onEnter={[Function]}
              onEntered={[Function]}
              onEntering={[Function]}
              onExit={[Function]}
              onExited={[Function]}
              onExiting={[Function]}
              timeout={
                Object {
                  "enter": 400,
                  "exit": 400,
                }
              }
              unmountOnExit={true}
            >
              <CartCount__Dot
                className="count"
              >
                <div
                  className="count CartCount__Dot-xxvp4g-1 jtvosm"
                >
                  10
                </div>
              </CartCount__Dot>
            </Transition>
          </CSSTransition>
        </div>
      </TransitionGroup>
    </span>
  </CartCount__AnimationStyles>
</CartCounter>
`;

exports[`<CartCount /> updates via props 2`] = `
<CartCounter
  count={50}
>
  <CartCount__AnimationStyles>
    <span
      className="CartCount__AnimationStyles-xxvp4g-0 gDkasO"
    >
      <TransitionGroup
        childFactory={[Function]}
        component="div"
      >
        <div>
          <CSSTransition
            className="count"
            classNames="count"
            in={true}
            key=".$50"
            onExited={[Function]}
            timeout={
              Object {
                "enter": 400,
                "exit": 400,
              }
            }
            unmountOnExit={true}
          >
            <Transition
              appear={false}
              className="count"
              enter={true}
              exit={true}
              in={true}
              mountOnEnter={false}
              onEnter={[Function]}
              onEntered={[Function]}
              onEntering={[Function]}
              onExit={[Function]}
              onExited={[Function]}
              onExiting={[Function]}
              timeout={
                Object {
                  "enter": 400,
                  "exit": 400,
                }
              }
              unmountOnExit={true}
            >
              <CartCount__Dot
                className="count"
              >
                <div
                  className="count CartCount__Dot-xxvp4g-1 jtvosm"
                >
                  50
                </div>
              </CartCount__Dot>
            </Transition>
          </CSSTransition>
          <CSSTransition
            className="count"
            classNames="count"
            in={false}
            key=".$10"
            onExited={[Function]}
            timeout={
              Object {
                "enter": 400,
                "exit": 400,
              }
            }
            unmountOnExit={true}
          >
            <Transition
              appear={false}
              className="count"
              enter={true}
              exit={true}
              in={false}
              mountOnEnter={false}
              onEnter={[Function]}
              onEntered={[Function]}
              onEntering={[Function]}
              onExit={[Function]}
              onExited={[Function]}
              onExiting={[Function]}
              timeout={
                Object {
                  "enter": 400,
                  "exit": 400,
                }
              }
              unmountOnExit={true}
            >
              <CartCount__Dot
                className="count"
              >
                <div
                  className="count CartCount__Dot-xxvp4g-1 jtvosm"
                >
                  10
                </div>
              </CartCount__Dot>
            </Transition>
          </CSSTransition>
        </div>
      </TransitionGroup>
    </span>
  </CartCount__AnimationStyles>
</CartCounter>
`;
